### ISSUE TITLE
Implement masking during softmax / softcap computation in GQA CPU operator

### ISSUE THREAD

**Issue by derdeljan-msft** [microsoft/onnxruntime#23982]
In https://github.com/microsoft/onnxruntime/pull/23944, custom causal attention mask is applied as a separate pass before softmax / softcap computation in GQA and adds attention mask values to the logits. Optimal solution would be to apply the attention mask during reduce computation in softmax / softcap.

cc @fajin-corp
